packages/next/build/webpack-config.ts

```diff
@@ -760,7 +776,9 @@ export default async function getBaseWebpackConfig(
       rules: [
         {
           test: /\.(tsx|ts|js|mjs|jsx)$/,
-          include: [dir, ...babelIncludeRegexes],
+          include: resolvedBaseUrl
+            ? [resolvedBaseUrl, ...babelIncludeRegexes]
+            : [dir, ...babelIncludeRegexes],
```

Comment:
Not sure I understand this change.

Reply:
Without this specific change, babel-loader won't transpile files from the mono repo which resolve outside of the next app directory through the `paths` config.

This is particularly important in the case of the example repository I provided which has the following structure:
```
.
└── packages
    ├── bar
    ├── baz
    │   └── src
    ├── foo
    └── web-app
```
In this example when running `yarn --cwd packages/web-app next build`:
`dir` is the directory where `next.config.js` is, which resolves to `./packages/web-app`.
using this as include, prevents babel-loader from transpiling files in `./packages/bar`,`./packages/foo` and `./packages/baz`. Thus when they are resolved through the `paths` configuration in tsconfig, webpack will try to load them directly without transpilation.

Reply:
This doesn't handle the case that a user may specify `baseUrl` as `node_modules`, effectively disabling all compilation for your project.

Reply:
I can't see a case where that would make sense but I have updated the PR to account for this, I always include `dir` but I also include `resolvedBaseUrl` if is is defined.